allauth/socialaccount/providers/agave/provider.py

```diff
@@ -26,8 +26,9 @@ def extract_uid(self, data):
     def extract_common_fields(self, data):
         return dict(
             email=data.get('email'),
-            username=data.get('username'),
-            name=(data.get('first_name') + ' ' + data.get('last_name')),
+            username=data.get('username', ''),
+            name=(data.get('first_name', '') +
+                  ' ' + data.get('last_name', '')),
```

Comment:
If either last/first name is blank, it returns in a trailing/leading white space. Could use a strip()